engine/src/main/java/io/camunda/zeebe/engine/state/analyzers/CatchEventAnalyzer.java

```diff
@@ -201,15 +208,20 @@ private Optional<CatchEventTuple> findEscalationCatchEventInScope(
     final var elementType = processInstanceRecord.getBpmnElementType();
 
     final var element = process.getElementById(elementId, elementType, ExecutableActivity.class);
-    final Optional<ExecutableCatchEvent> catchEvent =
+    final Optional<ExecutableCatchEvent> escalstionCatchEvent =
         element.getEvents().stream()
             .filter(ExecutableCatchEvent::isEscalation)
+            // Because a catch event can not contain an expression, we ignore it if not set.
+            .filter(catchEvent -> catchEvent.getEscalation().getEscalationCode().isPresent())
+            // Order by escalationCode to prioritize code-specific escalation events within the same
+            // scope.
+            .sorted(ESCALATION_CODE_COMPARATOR)
             .filter(event -> matchesEscalationCode(event, escalationCode))
             .findFirst();
 
-    if (catchEvent.isPresent()) {
+    if (escalstionCatchEvent.isPresent()) {
```

Comment:
:wrench: Typo here as well:

```suggestion
    if (escalationCatchEvent.isPresent()) {
```